go/cmd/vtgateclienttest/services/errors.go

```diff
@@ -36,58 +43,218 @@ func newErrorClient(fallback vtgateservice.VTGateService) *errorClient {
 	}
 }
 
+// requestToError returns an error for the given request, by looking at the
+// request's prefix and requested error type. If the request doesn't match an
+// error request, return nil.
+func requestToError(request string) error {
+	if !strings.HasPrefix(request, ErrorPrefix) {
+		return nil
+	}
+	return trimmedRequestToError(strings.TrimPrefix(request, ErrorPrefix))
+}
+
+// requestToPartialError attempts to return a partial error for a given request.
```

Comment:
I'm struggling to understand what a partial error is, so this circular definition is grating.